spack/configs/summit/packages.yaml

```diff
@@ -2,16 +2,27 @@ packages:
 
   # concretization preferences
   all:
-    compiler: [gcc@8.1.1]
     providers:
       mpi: [spectrum-mpi]
+      blas: [netlib-lapack]
+      lapack: [netlib-lapack]
+      scalapack: [netlib-scalapack]
+
+  gene:
+    compiler: [gcc@8.1.1]
+
+  xgc1:
+    compiler: [pgi@19.9]
```

Comment:
This isn't working for me, the default xgc1 spec is still to use `gcc@8.1.1`. This might be because a bunch of it's dependencies are already installed with gcc for gene. The 'which version should I use by default' question is a complicated one - for externals, I generally want it to heavily favor pre-installed versions, in many cases even if there is a compiler mismatch (which can be achieved by not specifiying a compiler for the external even when known, which is what I have been doing). In this case, it seems like it should be obvious - I explicitly said I want to build xgc1 with pgi@19.9 and nothing else, so all it's deps should follow.

Reply:
I have no idea why this doesn't work, either -- I'd have thought it should work. And it'd be useful to be able to set such defaults (though unless there's a specific reason not to, I'd have gone with the latest pgi).